src/main/java/com/labsynch/labseer/chemclasses/rdkit/CmpdRegMoleculeRDKitImpl.java

```diff
@@ -0,0 +1,195 @@
+package com.labsynch.labseer.chemclasses.rdkit;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.labsynch.labseer.chemclasses.CmpdRegMolecule;
+import com.labsynch.labseer.exceptions.CmpdRegMolFormatException;
+import com.labsynch.labseer.utils.PropertiesUtilService;
+import com.labsynch.labseer.utils.SimpleUtil;
+
+import org.RDKit.RDKFuncs;
+import org.RDKit.ROMol;
+import org.RDKit.RWMol;
+import org.RDKit.Str_Vect;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.node.ObjectNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Configurable;
+
+@Configurable
+public class CmpdRegMoleculeRDKitImpl implements CmpdRegMolecule {
+
+	Logger logger = LoggerFactory.getLogger(CmpdRegMoleculeRDKitImpl.class);
+
+	@Autowired
+	private PropertiesUtilService propertiesUtilService;
+
+	RWMol molecule;
+
+	static {
+		System.loadLibrary("GraphMolWrap");
+	}
+
+	public CmpdRegMoleculeRDKitImpl(String molStructure) throws CmpdRegMolFormatException {
+		this.molecule = RWMol.MolFromMolBlock(molStructure);
+	}
+
+	public CmpdRegMoleculeRDKitImpl(RWMol molecule) throws CmpdRegMolFormatException {
+		this.molecule = molecule;
+	}
+
+	public CmpdRegMoleculeRDKitImpl(ROMol readOnlyMol) {
+		if (readOnlyMol == null) {
+			logger.info("Got empty mol record");
+		} else {
+			this.molecule = RWMol.MolFromMolBlock(readOnlyMol.MolToMolBlock());
+		}
+	}
+
+	@Override
+	public void setProperty(String key, String value) {
+		molecule.setProp(key, value);
+	}
+
+	@Override
+	public String getProperty(String key) {
+		return molecule.getProp(key);
+	}
+
+	@Override
+	public String[] getPropertyKeys() {
+		if (this.molecule != null) {
+			List<String> propertyKeys = new ArrayList<String>();
+			Str_Vect propList = this.molecule.getPropList();
+			for (int i = 0; i < propList.size(); i++) {
+				propertyKeys.add(propList.get(i));
+			}
+			String[] keys = new String[propertyKeys.size()];
+			return propertyKeys.toArray(keys);
+		} else {
+			return new String[0];
+		}
+	}
+
+	@Override
+	public String getPropertyType(String key) {
+		// Does not exist
```

Comment:
Is this being used anywhere?

Reply:
It does actually get returned all the way back to the client as part of its call to `/api/cmpdRegBulkLoader/readSDF`, however reading through that code, its not actually used.  Just the names.  This is the same impl as the indigo impl so I think its fine to leave it this way.